x-pack/plugins/observability_solution/infra/public/alerting/inventory/rule_data_formatters.ts

```diff
@@ -7,13 +7,27 @@
 
 import { ALERT_REASON } from '@kbn/rule-data-utils';
 import { ObservabilityRuleTypeFormatter } from '@kbn/observability-plugin/public';
+import type { LocatorPublic } from '@kbn/share-plugin/common';
+import type {
+  AssetDetailsLocatorParams,
+  InventoryLocatorParams,
+} from '@kbn/observability-shared-plugin/common';
 import { getInventoryViewInAppUrl } from '../../../common/alerting/metrics/alert_link';
 
-export const formatReason: ObservabilityRuleTypeFormatter = ({ fields }) => {
-  const reason = fields[ALERT_REASON] ?? '-';
+export const getFormatReason = ({
```

Comment:
nit: Maybe renaming it to `ruleFormat` would be more clear as it also returns viewInApp link beside formatting the reason.